src/firefly/js/visualize/HiPSUtil.js

```diff
@@ -0,0 +1,344 @@
+    const dir= Math.floor(tileNumber/10000)*10000;
+    return `${plot.hipsUrlRoot}/Norder${nOrder}/Dir${dir}/Npix${tileNumber}.${getHiPSTileExt(plot)}`;
+}
+
+export function makeHiPSAllSkyUrl(plot) {
+    if (!plot) return null;
+    return `${plot.hipsUrlRoot}/Norder3/Allsky.${getHiPSTileExt(plot)}`;
+}
+
+export function getHiPSTileExt(plot) {
+    const exts= get(plot, 'hipsProperties.hips_tile_format', 'jpg');
+    if (!plot) return null;
+    if (exts.includes('png')) return'png';
+    else if (exts.includes('jpeg') || exts.includes('jpg')) return 'jpg';
+    else return 'jpg';
+
+}
+
+const offsetDeg= 72;
+/**
+ *
+ * @param plot
+ * @param viewDim
+ * @return {{centerWp:WorldPt, fov: number, centerDevPt: DevicePt}}
+ */
+export function getPointMaxSide(plot, viewDim) {
+
+    const cc= CysConverter.make(plot);
+    const {width,height}= viewDim;
+    const centerDevPt= makeDevicePt(width/2, height/2);
+    const centerWp= cc.getWorldCoords( centerDevPt, plot.imageCoordSys);
+
+    const pt0= cc.getWorldCoords( makeDevicePt(0,0), plot.imageCoordSys);
+    const ptWidth= cc.getWorldCoords( makeDevicePt(width,0), plot.imageCoordSys);
+    const ptHeight= cc.getWorldCoords( makeDevicePt(0,height), plot.imageCoordSys);
+
+
+
+
+    if (!ptWidth || !ptHeight) {
+        const newLat= (centerWp.y>0) ? centerWp.y-offsetDeg : centerWp.y+offsetDeg;
+        // const devRadEdge= cc.getDeviceCoords( makeWorldPt(centerWp.x, newLat, plot.imageCoordSys));
+        // const devRadius= Math.abs(centerDevPt.y- devRadEdge.y);
+        // const devRadius= Math.abs(centerDevPt.y- devRadEdge.y);
+        // console.log(devRadius);
+        return {centerWp,centerDevPt, fov:180};
+    }
+
+    const widthInDeg= computeDistance(pt0, ptWidth);
+    const heightInDeg= computeDistance(pt0, ptHeight);
+    return {centerWp, centerDevPt, fov:Math.max(widthInDeg,heightInDeg)};
+}
+
+
+/**
+ *
+ * @param {PlotView} pv
+ * @param {number} size in degrees
+ * @return {number} a zoom level
+ */
+export function getHiPSZoomLevelToFit(pv,size) {
+    const {width,height}=pv.viewDim;
+    const plot= primePlot(pv);
+    if (!plot || !width || !height) return 1;
+
+    // make version of plot centered at 0,0
+    const tmpPlot= changeProjectionCenter(plot, makeWorldPt(0,0, plot.imageCoordSys));
+    const cc= CysConverter.make(tmpPlot)
+    const pt1= cc.getImageCoords( makeWorldPt(0,0));
+    const pt2= cc.getImageCoords( makeWorldPt(size,0));
+    return Math.min(width, height)/Math.abs(pt2.x-pt1.x);
+}
+
+/**
+ *
+ * @param {PlotView} pv
+ * @return {number} fov in degrees
+ */
+export function getHiPSFoV(pv) {
+    const cc= CysConverter.make(primePlot(pv));
+    const {width,height}=pv.viewDim;
+    if (!cc || !width || !height) return;
+
+    const pt1= cc.getWorldCoords( makeDevicePt(1,height/2));
+    const pt2= cc.getWorldCoords( makeDevicePt(width-1,height/2));
+    return (pt1 && pt2) ? computeDistance(pt1,pt2) : 180;
+}
+
+
+const HealpixCache = {
```

Comment:
Also, are we planning to have one HEALPix image at any time?

Reply:
This is a idea taken from Aladin. I have not done any testing to determine if this is necessary.  It is part of what I want to do next.  My plans are probably to take it out but I need to do more testing.

Reply:
More on this.  The code is way to complex so I rewrote it to a much simpler version. Basically we have to draw a lot tiles when we are at level 3 which is usually show the whole sky (nside 8). The array of corners for level 3 is cached. Since HiPS tile are the same for any image cacheing level three corners makes a lot of sense.